ark/type/src/parser/tuple.ts

```diff
@@ -52,7 +53,8 @@ export const parseTupleLiteral = (def: List, ctx: ParseContext): TypeNode => {
 			elementDef = elementDef[1]
 			isVariadic = true
 		}
-		const value = ctx.scope.parse(elementDef, ctx)
+		const keyData = checkForOptionalAndTrimData(elementDef)
```

Comment:
Let's reuse the `parseKey` strategy I mentioned earlier here as well. Maybe something like `parseEntry`, it accepts a key and a value and returns the information similar to the parsedKey type.

For an array, the key can just be the index and it will still work for parsing the value/extracting the inner def etc.